File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = {}

        for n in nums:
            mp[n] = mp.get(n, 0) + 1

        l = list(mp.keys())

        l.sort(key=lambda x: mp[x], reverse=True)

        return l[:k]

File: Data Structures & Algorithms/house-robber-ii/submission-0.java
class Solution {
    public int rob(int[] nums) {
        int n = nums.length;

        if (n == 1) return nums[0];

        return Math.max(
            robRange(nums, 0, n - 2),
            robRange(nums, 1, n - 1)
        );
    }

    private int robRange(int[] nums, int start, int end) {
        int[] dp = new int[nums.length + 2];

        for (int i = end; i >= start; i--) {
            dp[i] = Math.max(
                nums[i] + dp[i + 2],
                dp[i + 1]
            );
        }

        return dp[start];
    }
}